File: Java Code/快速排序.java
/**
 * Created by Lee on 2018/7/23.
 */
public class Quick {
    public static void QuickSort(int a[],int low ,int high){
        int i,j,key;
        i = low;j = high;
        if(low<high) {
            key = a[low];
            while (i < j) {
                while (i < j && a[j] >= key)
                    j--;
                if (i < j)
                    a[i] = a[j];

                while (i < j && a[i] <= key)
                    i++;
                if (i < j)
                    a[j] = a[i];
            }
            a[i] = key;
            QuickSort(a, low, i - 1);
            QuickSort(a, i + 1, high);
        }
    }

    public static void main(String args[]){
        int[] a ={5,1,3,4,2,9};
        QuickSort(a,0,a.length-1);
        for(int num:a)
            System.out.print(num+" ");
    }
}

File: C  Code/快速排序.cpp
#include <stdio.h>
#include <stdlib.h>


typedef struct
{
	int data[100];
	int length;
}SqList;

void Out(SqList *L)
{
	int i;
	for (i = 0; i < L->length; i++)
	{
		printf("%d ", L->data[i]);
	}
}
void Insert(SqList *L,int size)
{
	int i;
	L->length = size;
	printf("输入%d个元素", size );
	for (i = 0; i < L->length; i++)
	{
		scanf_s("%d", &L->data[i]);
	}
}

void Quick(SqList *L,int low ,int high)
{
	int i , j , key;
	if (low < high)
	{
		key = L->data[low];
		i = low;
		j = high;
		while (i < j)
		{
			while (L->data[j] >= key && i < j)
				j--;
			if (i < j)
				L->data[i++] = L->data[j];
			while (L->data[i] <= key && i < j)
				i++;
			if (i < j)
				L->data[j--] = L->data[i];
		}
		L->data[i] = key;
		Quick(L, low, i - 1);
		Quick(L, i + 1, high);
	}

}

int main()
{
	int size;
	SqList L;
	printf("输入表的长度:");
	scanf_s("%d", &size);
	Insert(&L,size);
	Quick(&L,0,size-1);
	printf("\n快速排序后的元素:");
	Out(&L);
}